### PR TITLE
Docs: Add guide for sharing plugins

### PR DESCRIPTION
**Pull request by marcusolsson**
**What this PR does / why we need it**:

Adds a guide for how to share plugins. It tries to explain:

- How Grafana loads plugins
- How to submit a plugin to Grafana.com
- How to host/distribute a plugin yourself

This will replace the [Publish your plugin](https://grafana.com/tutorials/build-a-panel-plugin/#8) step in the plugin tutorials. Since publishing plugins isn't part of getting started, nor will everyone do it, it's better suited as a guide.

### CODE REVIEW

File: docs/sources/developers/plugins/share-a-plugin.md [hunk 1]
```diff
@@ -0,0 +1,122 @@
...
+
+# Share a plugin
+
+You've just built your first plugin and now you want to share it with the world. In this guide, you'll learn how to package and share your plugin with others.
+
+When you build a plugin from source, a `dist` directory is created that contains the production build, or _plugin assets_, for your plugin.
+
+When loading your plugin, Grafana only cares about the plugin assets. Specifically, when the Grafana server starts, it attempts to discover and load plugins like this:
+
+1. Look for for a `plugin.json` file in any of the subdirectories in the plugin directory
```
Comment (briangann) [inline docs/sources/developers/plugins/share-a-plugin.md:14]: there's a duplicate "for" in this sentence

**Review by briangann — CHANGES_REQUESTED** [review]
one typo, looks great otherwise 👍

File: docs/sources/menu.yaml [hunk 2]
```diff
@@ -260,26 +260,26 @@
...
-        name: Add interval variable
-      - link: /variables/variable-types/add-ad-hoc-filters/
-        name: Add ad hoc filters
-      - link: /variables/variable-types/chained-variables/
-        name: Chained variables
-      - link: /variables/variable-types/global-variables/
-        name: Global variables
-      - link: /variables/variable-types/url-variables/
-        name: URL variables
+        - link: /variables/variable-types/add-query-variable/
```
Comment (oddlittlebird) [inline docs/sources/menu.yaml:263]: @marcusolsson, are these indents on purpose? The menu was building correctly before.

File: docs/sources/developers/plugins/share-a-plugin.md [hunk 3]
```diff
@@ -0,0 +1,122 @@
++++
+title = "Share a plugin"
+type = "docs"
++++
+
+# Share a plugin
+
+You've just built your first plugin and now you want to share it with the world. In this guide, you'll learn how to package and share your plugin with others.
```
Comment (oddlittlebird) [inline docs/sources/developers/plugins/share-a-plugin.md:8]:
```suggestion
You've just built your first plugin, and now you want to share it with the world. In this guide, you'll learn how to package and share your plugin with others.
```

File: docs/sources/developers/plugins/share-a-plugin.md (hunk 1 above)
Comment (oddlittlebird) [reply docs/sources/developers/plugins/share-a-plugin.md:14]: Please put periods at the end of each sentence in this list.

File: docs/sources/developers/plugins/share-a-plugin.md [hunk 4]
```diff
@@ -0,0 +1,122 @@
...
+
+When you build a plugin from source, a `dist` directory is created that contains the production build, or _plugin assets_, for your plugin.
+
+When loading your plugin, Grafana only cares about the plugin assets. Specifically, when the Grafana server starts, it attempts to discover and load plugins like this:
+
+1. Look for for a `plugin.json` file in any of the subdirectories in the plugin directory
+1. If a `plugin.json` was found, try to load the plugin assets from a `dist` directory in the same directory as the `plugin.json` file
+1. If there's no `dist` directory, try to load the plugin assets from the same directory as the `plugin.json` file
+
+Now that you know what Grafana needs to load your plugin, let's see how you can share the plugin with your users.
```
Comment (oddlittlebird) [inline docs/sources/developers/plugins/share-a-plugin.md:18]:
```suggestion
Now that you know what Grafana needs to load your plugin, let's see how you can share the plugin with other users.
```

Comment (oddlittlebird) [reply docs/sources/developers/plugins/share-a-plugin.md:18]: Do we mean "users" or do we mean "with the community"?

File: docs/sources/developers/plugins/share-a-plugin.md [hunk 5]
```diff
@@ -0,0 +1,122 @@
...
+
+The best way to share your plugin with the world is to publish it on [Grafana Plugins](https://grafana.com/plugins). However, if you're not ready to make your plugin public just yet, you can still share your plugin by hosting the plugin yourself.
+
+## Publish your plugin on Grafana.com
+
+To publish a plugin to [Grafana Plugins](https://grafana.com/grafana/plugins), your plugin first needs to be publicly available in a commit on [GitHub](https://github.com).
+
+The commit you submit needs to either:
+
+- contain a `dist` directory with the plugin assets
```
Comment (oddlittlebird) [inline docs/sources/developers/plugins/share-a-plugin.md:28]:
```suggestion
- Contain a `dist` directory with the plugin assets
```

File: docs/sources/developers/plugins/share-a-plugin.md [hunk 6]
```diff
@@ -0,0 +1,122 @@
...
+- contain the plugin assets in the root directory
```
Comment (oddlittlebird) [inline docs/sources/developers/plugins/share-a-plugin.md:29]:
```suggestion
- Contain the plugin assets in the root directory
```

File: docs/sources/developers/plugins/share-a-plugin.md [hunk 7]
```diff
@@ -0,0 +1,122 @@
...
+
+We strongly recommend that you don't check in the plugin assets to the main branch. Instead, use the following steps to create a release branch that contains the plugin assets.
+
+1. Create a release branch
```
Comment (oddlittlebird) [inline docs/sources/developers/plugins/share-a-plugin.md:33]:
```suggestion
1. Create a release branch.
```

File: docs/sources/developers/plugins/share-a-plugin.md [hunk 8]
```diff
@@ -0,0 +1,122 @@
...
+   git add -f dist
+   ```
+
+1. Create the release commit.
+
+   ```
+   git commit -m "Release v0.1.0"
+   ```
+
+1. Create a release tag. You can also [create the release on GitHub](https://docs.github.com/en/github/administering-a-repository/managing-releases-in-a-repository). If you do, you can skip this step, as well as the next one.
```
Comment (oddlittlebird) [inline docs/sources/developers/plugins/share-a-plugin.md:57]:
```suggestion
1. Create a release tag. You can also [create the release on GitHub](https://docs.github.com/en/github/administering-a-repository/managing-releases-in-a-repository). If you do, then you can skip this step and the next one.
```

File: docs/sources/developers/plugins/share-a-plugin.md [hunk 9]
```diff
@@ -0,0 +1,122 @@
...
+
+The next step is to submit the URL to your repository, and the release commit, to the [Grafana Plugin Repository](https://github.com/grafana/grafana-plugin-repository).
+
+## Host the plugin yourself
+
+If you want to share your plugin by hosting it yourself, we recommend that you package it by adding the plugin assets to a ZIP archive. You can then make the archive available by hosting them yourself.
+
+How you package the plugin depends on whether you want to include the source code or not.
+
+
```
Comment (oddlittlebird) [inline docs/sources/developers/plugins/share-a-plugin.md:77]: Extra lines in here. Please remove all but one.

File: docs/sources/developers/plugins/share-a-plugin.md [hunk 10]
```diff
@@ -0,0 +1,122 @@
...
+   ```
+   git push --set-upstream origin release-0.1.x --follow-tags
+   ```
+
+The next step is to submit the URL to your repository, and the release commit, to the [Grafana Plugin Repository](https://github.com/grafana/grafana-plugin-repository).
+
+## Host the plugin yourself
+
+If you want to share your plugin by hosting it yourself, we recommend that you package it by adding the plugin assets to a ZIP archive. You can then make the archive available by hosting them yourself.
```
Comment (oddlittlebird) [inline docs/sources/developers/plugins/share-a-plugin.md:73]:
```suggestion
If you want to share your plugin by hosting it yourself, then we recommend that you package it by adding the plugin assets to a ZIP archive. You can then make the archive available by hosting it yourself.
```

File: docs/sources/developers/plugins/share-a-plugin.md [hunk 11]
```diff
@@ -0,0 +1,122 @@
...
+
+How you package the plugin depends on whether you want to include the source code or not.
+
+
+
+### Package the plugin with source code
+
+If you want to distribute the source code along with your plugin assets, you can archive the entire plugin directory.
```
Comment (oddlittlebird) [inline docs/sources/developers/plugins/share-a-plugin.md:81]:
```suggestion
If you want to distribute the source code along with your plugin assets, then you can archive the entire plugin directory.
```

File: docs/sources/developers/plugins/share-a-plugin.md [hunk 12]
```diff
@@ -0,0 +1,122 @@
...
+
+To create a ZIP archive that contains the plugin assets and source code, run the following commands in your terminal:
```
Comment (oddlittlebird) [inline docs/sources/developers/plugins/share-a-plugin.md:83]:
```suggestion
To create a .zip archive that contains the plugin assets and source code, run the following commands in your terminal:
```

File: docs/sources/developers/plugins/share-a-plugin.md [hunk 13]
```diff
@@ -0,0 +1,122 @@
...
+
+```
+cd my-plugin/
+yarn build
+zip my-plugin-0.2.0.src.zip . -r -x "node_modules/*" -x ".git/*"
+```
+
+### Package the plugin without source code
+
+If you don't want to distribute the plugin with the source code, you can instead archive the `dist` directory.
```
Comment (oddlittlebird) [inline docs/sources/developers/plugins/share-a-plugin.md:93]:
```suggestion
If you don't want to distribute the plugin with the source code, then you can archive the `dist` directory.
```

File: docs/sources/developers/plugins/share-a-plugin.md [hunk 14]
```diff
@@ -0,0 +1,122 @@
...
+
+To create a ZIP archive that only contains the bare minimum to load the plugin, run the following commands in your terminal:
```
Comment (oddlittlebird) [inline docs/sources/developers/plugins/share-a-plugin.md:95]:
```suggestion
To create a .zip archive that only contains the bare minimum to load the plugin, run the following commands in your terminal:
```

Comment (oddlittlebird) [reply docs/sources/developers/plugins/share-a-plugin.md:95]: Per the MSG, file extensions are all lowercase and preceded by a period. https://docs.microsoft.com/en-us/style-guide/procedures-instructions/formatting-text-in-instructions

File: docs/sources/developers/plugins/share-a-plugin.md [hunk 15]
```diff
@@ -0,0 +1,122 @@
...
+```
+cd my-plugin/
+yarn build
+cd dist/
+zip my-plugin-0.2.0.nosrc.zip . -r
+```
+
+### Package and host your plugin using GitHub
+
+If your plugin is hosted on GitHub, you can share the plugin using the following URL:
```
Comment (oddlittlebird) [inline docs/sources/developers/plugins/share-a-plugin.md:106]:
```suggestion
If you host your plugin on GitHub, then you can share the plugin using the following URL:
```

File: docs/sources/developers/plugins/share-a-plugin.md [hunk 16]
```diff
@@ -0,0 +1,122 @@
...
+https://github.com/GITHUB_USERNAME/GITHUB_REPO_NAME/archive/<VERSION>.zip
+```
+
+For example, you can download the [Worldmap Panel](https://github.com/grafana/worldmap-panel) using the following URL:
+
+[https://github.com/grafana/worldmap-panel/archive/v0.3.2.zip](https://github.com/grafana/worldmap-panel/archive/v0.3.2.zip)
+
+### Install a packaged plugin
+
+Once the user has downloaded the archive containing the plugin assets, they can install it by extracting the archive into their plugin directory.
```
Comment (oddlittlebird) [inline docs/sources/developers/plugins/share-a-plugin.md:118]:
```suggestion
After the user has downloaded the archive containing the plugin assets, they can install it by extracting the archive into their plugin directory.
```

**Review by oddlittlebird — CHANGES_REQUESTED** [review]

File: docs/sources/menu.yaml (hunk 2 above)
Comment (marcusolsson) [reply docs/sources/menu.yaml:263]: Yes. While there's no semantic difference, the indentations were inconsistent with the rest of the document.

Compare

```
- name: Developers
  children:
    - name: Plugins
``` 

with

```
- name: Developers
  children:
  - name: Plugins
```

**Review by oddlittlebird — APPROVED** [review]
Looks good. Great content!

**Review by briangann — APPROVED** [review]
lgtm!

**Pull request merged**